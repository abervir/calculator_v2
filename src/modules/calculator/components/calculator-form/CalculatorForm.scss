.calculator-form {
  .inputs {
    display: flex;
    align-items: flex-end;
    height: 42px;

    .calc-input {
      input {
        font-size: 20px;
      }
    }

    > * {
      width: 100%;
    }

    .calculator-operations-list {
      margin-left: 5px;
      margin-right: 5px;
    }
  }

  .button {
    width: 100%;
    margin-top: 10px;
  }
}
