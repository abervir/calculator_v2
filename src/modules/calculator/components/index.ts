export * from './calculator-form/CalculatorForm';
export * from './calculator-operation-list/CalculatorOperationsList';
export * from './calculation-results-list/CalculationResultsList';
export * from './calculation-results-list-item/CalculationResultsListItem';
