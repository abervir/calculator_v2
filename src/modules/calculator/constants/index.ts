export * from './calculator-operations.constant';
