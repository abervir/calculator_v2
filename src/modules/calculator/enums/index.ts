export * from './operation-icons.enum';
export * from './math-operation.enum';
