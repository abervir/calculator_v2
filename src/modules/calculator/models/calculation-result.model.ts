import { CalculatorOperation } from './calculator-operation.model';

export interface CalculationResult {
  id: number;
  result: number | string;
  operands: number[];
  operation: CalculatorOperation;
}
