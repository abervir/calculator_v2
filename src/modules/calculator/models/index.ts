export * from './calculator-operation.model';
export * from './calculation-result.model';
