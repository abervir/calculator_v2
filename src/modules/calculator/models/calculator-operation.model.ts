import { OperationIcons } from '../enums';

export interface CalculatorOperation {
  id: string;
  title: string;
  icon: OperationIcons;
  sign: string;
  operation: (a: number, b: number) => number | null;
}
