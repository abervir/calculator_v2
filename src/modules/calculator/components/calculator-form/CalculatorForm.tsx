import React from 'react';

import { Button, Input } from '@modules/shared/components';
import { useCalculatorForm } from '@modules/calculator/hooks';
import { CalculationResult } from '@modules/calculator/models';
import { CalculatorOperationsList } from '../calculator-operation-list/CalculatorOperationsList';

import './CalculatorForm.scss';

interface Props {
  onSumbit: (calculationResult: CalculationResult) => void;
}

export const CalculatorForm = (props: Props) => {
  const { onCalculate, onChangeOperation, onSetOperands } = useCalculatorForm(
    props.onSumbit
  );

  return (
    <div className='calculator-form'>
      <div className='inputs'>
        <Input
          placeholder='Operand 1'
          isNumbersOnly
          onChange={(value) => onSetOperands(0, Number(value))}
        />
        <CalculatorOperationsList onChangeOperation={onChangeOperation} />
        <Input
          placeholder='Operand 2'
          isNumbersOnly
          onChange={(value) => onSetOperands(1, Number(value))}
        />
      </div>
      <Button title='Calculate' onClick={onCalculate} />
    </div>
  );
};
