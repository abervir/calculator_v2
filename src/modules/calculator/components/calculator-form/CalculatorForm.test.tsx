import { CalculationResult } from '@modules/calculator/models';
import { shallow } from 'enzyme';
import React from 'react';

import { CalculatorForm } from './CalculatorForm';

describe('CalculatorForm', () => {
  it('should render component', () => {
    expect(
      shallow(<CalculatorForm onSumbit={(_: CalculationResult) => {}} />)
    ).toMatchSnapshot();
  });
});
