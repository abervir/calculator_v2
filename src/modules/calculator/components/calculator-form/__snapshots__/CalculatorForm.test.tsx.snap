// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorForm should render component 1`] = `
<div
  className="calculator-form"
>
  <div
    className="inputs"
  >
    <Input
      isNumbersOnly={true}
      onChange={[Function]}
      placeholder="Operand 1"
    />
    <CalculatorOperationsList
      onChangeOperation={[Function]}
    />
    <Input
      isNumbersOnly={true}
      onChange={[Function]}
      placeholder="Operand 2"
    />
  </div>
  <Button
    onClick={[Function]}
    title="Calculate"
  />
</div>
`;
